File: scripts/restore-files.js
import { execSync } from 'child_process';
import { mkdirSync, writeFileSync, existsSync } from 'fs';
import { dirname, join } from 'path';

const PROJECT_ROOT = '/vercel/share/v0-project';
const GIT_DIR = join(PROJECT_ROOT, '.git');

// Use --git-dir to specify the git directory explicitly
const gitCmd = `git --git-dir=${GIT_DIR} --work-tree=${PROJECT_ROOT}`;

try {
  // Get all tracked files in git under src/
  const gitFiles = execSync(`${gitCmd} ls-tree -r HEAD --name-only`, { encoding: 'utf-8' })
    .trim()
    .split('\n')
    .filter(f => f.startsWith('src/'));

  console.log(`Found ${gitFiles.length} files in git under src/`);

  let restored = 0;
  let alreadyExist = 0;
  let errors = 0;

  for (const file of gitFiles) {
    const fullPath = join(PROJECT_ROOT, file);
    try {
      // Check if file physically exists on disk
      if (existsSync(fullPath)) {
        alreadyExist++;
        continue;
      }

      // Get file content from git
      const content = execSync(`${gitCmd} show HEAD:${file}`, { encoding: 'utf-8' });
      
      // Ensure directory exists
      const dir = dirname(fullPath);
      mkdirSync(dir, { recursive: true });
      
      // Write file
      writeFileSync(fullPath, content, 'utf-8');
      restored++;
      console.log(`Restored: ${file}`);
    } catch (err) {
      errors++;
      console.error(`Error restoring ${file}: ${err.message}`);
    }
  }

  console.log(`\nDone! Restored: ${restored}, Already existed: ${alreadyExist}, Errors: ${errors}`);
} catch (err) {
  console.error('Fatal error:', err.message);
  
  // Fallback: try git checkout
  console.log('\nTrying fallback: git checkout...');
  try {
    execSync(`${gitCmd} checkout HEAD -- src/`, { encoding: 'utf-8', cwd: PROJECT_ROOT });
    console.log('Successfully checked out src/ from git!');
  } catch (err2) {
    console.error('Fallback also failed:', err2.message);
  }
}

File: scripts/restore-files.py
import os
import subprocess

# The v0 project root
PROJECT_ROOT = "/vercel/share/v0-project"

# Find all files that the Read tool can access but might not be on the filesystem
# Use find to see what actually exists in src/
src_path = os.path.join(PROJECT_ROOT, "src")

# Check if src directory exists
if os.path.isdir(src_path):
    existing = []
    for root, dirs, files in os.walk(src_path):
        for f in files:
            existing.append(os.path.relpath(os.path.join(root, f), PROJECT_ROOT))
    print(f"Found {len(existing)} files already on disk in src/:")
    for f in existing:
        print(f"  {f}")
else:
    print("src/ directory does not exist on disk!")
    os.makedirs(src_path, exist_ok=True)
    print("Created src/ directory")

# Try to use git if available
try:
    result = subprocess.run(
        ["git", "ls-tree", "-r", "HEAD", "--name-only"],
        capture_output=True, text=True, cwd=PROJECT_ROOT
    )
    if result.returncode == 0:
        git_files = [f for f in result.stdout.strip().split("\n") if f.startswith("src/")]
        print(f"\nGit reports {len(git_files)} files under src/")
        
        # Try git checkout for all src files
        checkout_result = subprocess.run(
            ["git", "checkout", "HEAD", "--", "src/"],
            capture_output=True, text=True, cwd=PROJECT_ROOT
        )
        if checkout_result.returncode == 0:
            print("Successfully checked out all src/ files from git!")
        else:
            print(f"Git checkout failed: {checkout_result.stderr}")
    else:
        print(f"Git ls-tree failed: {result.stderr}")
except FileNotFoundError:
    print("Git is not available")

# Verify what's there now
if os.path.isdir(src_path):
    count = 0
    for root, dirs, files in os.walk(src_path):
        count += len(files)
    print(f"\nAfter restore: {count} files in src/")
